soc/arm/quicklogic_eos_s3/fpga_loader.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * ==========================================================
+ *
+ *   Copyright 2020 QuickLogic
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ *
+ *    File   : fpga_loader.c
+ *    Purpose: Contains functionality to load FPGA
+ *
+ *=========================================================
+ */
+
+
+#include <zephyr.h>
+#include <soc.h>
+
+#define CFG_CTL_CFG_DATA                (*(volatile u32_t *)(0x40014FFC))
+#define CFG_CTL_CFG_CTL                 (*(volatile u32_t *)(0x40014000))
+
+
+int load_fpga(u32_t img_size, u32_t *image_ptr)
+{
+	unsigned int    i = 0;
+	u32_t        chunk_cnt = 0;
+	volatile u32_t   *gFPGAPtr = (volatile u32_t *)image_ptr;
+
+	IO_MUX->PAD_19_CTRL = 0x00000180;
+
+	enable_fpga_clocks();
+
+	/* wait some time for fpga to get reset pulse */
+#if 0
+	k_sleep(100);
+#else
+	for (i = 0; i < 60; i++)
```

Comment:
If we wait for reset pulse as stated above:
> /* wait some time for fpga to get reset pulse */

Maybe there is a kind of flag that could provide an information about it? If yes, then we could have just something like this:
```suggestion
	while (!REG->FLAG) {
		PMU->GEN_PURPOSE_1  = i << 4;
              k_sleep(1);
      }
```